packages/ra-core/src/form/ValidationError.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { SFC } from 'react';
+import {
+    ValidationErrorMessage,
+    ValidationErrorMessageWithArgs,
+} from './validate';
+import { useTranslate } from '../i18n';
+
+interface Props {
+    error: ValidationErrorMessage;
+}
+
+const ValidationError: SFC<Props> = ({ error }) => {
+    const translate = useTranslate();
+
+    if ((error as ValidationErrorMessageWithArgs).message) {
+        const { message, args } = error as ValidationErrorMessageWithArgs;
+        const { _, ...allArgsButDefault } = args;
+
+        const translatedArgs = Object.keys(allArgsButDefault).reduce(
```

Comment:
very functional but also very hard to read IMO. Could you rework that reduce to make it easier to read (hint: imperative style is sometimes better)?